### PR TITLE
[2.3][Process] fix Proccess run with pts enabled

### PR DESCRIPTION
**Pull request by ewgRa**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #12643 |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

**Comment by ewgRa** [comment]
When fabbot.io patch is applied, there is a conflicts, I think better to merge it without conflicts.

**Comment by ewgRa** [comment]
Test failure seems not related to this PR

File: src/Symfony/Component/Process/Process.php [hunk 1]
```diff
@@ -238,8 +238,21 @@ public function start($callback = null)
             }
         }
 
+        $ptsWorkaround = null;
+
+        if (!$this->useFileHandles && $this->enhanceSigchildCompatibility && $this->isSigchildEnabled()) {
+            // Workaround for the bug, when PTS functionality is enabled.
+            // @see : https://github.com/symfony/symfony/issues/12643
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Process/Process.php]: no need to ref this one in the code

File: src/Symfony/Component/Process/Process.php [hunk 2]
```diff
@@ -238,8 +238,21 @@ public function start($callback = null)
...
+            // @see : https://github.com/php/php-src/pull/1588
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Process/Process.php]: I'd ref https://bugs.php.net/69442 instead

**Comment by nicolas-grekas** [comment]
@ewgRa could you please fabbot's issues also (even if unrelated to your change)?

**Comment by nicolas-grekas** [comment]
See https://github.com/symfony/symfony/issues/12643#issuecomment-92279343 for some background on why this works

**Comment by nicolas-grekas** [comment]
:+1: (after rebasing)

**Comment by ewgRa** [comment]
done, anything else?

**Comment by nicolas-grekas** [comment]
Thank you @ewgRa.

**Comment by nicolas-grekas** [comment]
Closed via ab3c8f891a260a720ddb4dcc50fe721fdf30fe9f

**Pull request closed without merging**